x-pack/plugins/observability/server/lib/rules/threshold/register_threshold_rule_type.ts

```diff
@@ -161,47 +146,26 @@ export function thresholdRuleType(
     minimumLicenseRequired: 'basic' as LicenseType,
     isExportable: true,
     executor: createLifecycleRuleExecutor(
-      createMetricThresholdExecutor({ basePath, logger, config })
+      createMetricThresholdExecutor({ alertsLocator, basePath, logger, config })
     ),
     doesSetRecoveryContext: true,
     actionVariables: {
       context: [
-        { name: 'group', description: groupActionVariableDescription },
-        { name: 'groupByKeys', description: groupByKeysActionVariableDescription },
-        ...(getAlertDetailsPageEnabledForApp(config.unsafe.alertDetails, 'metrics')
-          ? [
-              {
-                name: 'alertDetailsUrl',
-                description: alertDetailUrlActionVariableDescription,
-                usesPublicBaseUrl: true,
-              },
-            ]
-          : []),
-        { name: 'alertState', description: alertStateActionVariableDescription },
+        { name: 'groupings', description: groupByKeysActionVariableDescription },
+        {
+          name: 'alertDetailsUrl',
+          description: alertDetailUrlActionVariableDescription,
+          usesPublicBaseUrl: true,
+        },
         { name: 'reason', description: reasonActionVariableDescription },
         { name: 'timestamp', description: timestampActionVariableDescription },
         { name: 'value', description: valueActionVariableDescription },
-        { name: 'metric', description: metricActionVariableDescription },
-        { name: 'threshold', description: thresholdActionVariableDescription },
         { name: 'cloud', description: cloudActionVariableDescription },
         { name: 'host', description: hostActionVariableDescription },
         { name: 'container', description: containerActionVariableDescription },
         { name: 'orchestrator', description: orchestratorActionVariableDescription },
         { name: 'labels', description: labelsActionVariableDescription },
         { name: 'tags', description: tagsActionVariableDescription },
-        { name: 'originalAlertState', description: originalAlertStateActionVariableDescription },
-        {
-          name: 'originalAlertStateWasALERT',
-          description: originalAlertStateWasActionVariableDescription,
-        },
-        {
-          name: 'originalAlertStateWasWARNING',
-          description: originalAlertStateWasActionVariableDescription,
-        },
-        {
-          name: 'originalAlertStateWasNO_DATA',
-          description: originalAlertStateWasActionVariableDescription,
-        },
```

Comment:
Are we removing these variables as we don't have `warning` and `no_data` in the threshold rule?

Reply:
Not completely, even if we had a `warning`, we still didn't want to have a boolean for the original state. 

When I talked to Chris, it seemed the issue that we were trying to solve was related to sending recovered action based on the original state somehow, so I created a separate ticket to discuss it with ResponseOps to see how it should be handled (https://github.com/elastic/kibana/issues/160984)

Reply:
FYI, this was the original request https://github.com/elastic/kibana/issues/145418 for which `context.originalAlertState` variables were introduced.

Reply:
Thanks Bena! :)